src/keydra/clients/splunk.py

```diff
@@ -245,74 +250,184 @@ def rotate_hectoken(self, inputname):
             )
         )
 
-    def rotate_hectoken_cloud(self, inputname):
+    def _get_splunkcloud_httpinput(self, inputname):
         '''
-        Rotate a Splunk HEC token on Splunk Cloud (Classic)
+        Get details for a Splunk HEC token on Splunk Cloud (Classic)
 
         :param inputname: The name of the HTTP input
         :type inputname: :class:`string`
 
-        :returns: New token value if successful
-        :rtype: :class:`string`
+        :returns: Entry details, or an empty list if input not found
+        :rtype: :class:`list`
+        '''
+        getresp = self._service.get(
+            '/services/dmc/config/inputs/__indexers/http/{}'.format(inputname),
+            output_mode='json'
+        )['body'].read()
 
-        https://docs.splunk.com/Documentation/SplunkCloud/8.2.2105/Admin/ManageHECtokens
+        return json.loads(getresp)['entry']
 
+    def _get_last_splunkcloud_deploytask(self):
         '''
-        response = self._service.get(
-            '/services/dmc/config/inputs/__indexers/http/{}'.format(inputname),
+        Gets the last deployment task to be run on Splunk Cloud (Classic)
+
+        :returns: Last deployment task Id
+        :rtype: :class:`string`
+
+        '''
+        taskresp = self._service.get(
+            '/services/dmc/deploy',
             output_mode='json'
-        )
-        if response.status != 200:
-            inputlist = self._service.get(
-                '/services/dmc/config/inputs/-/http',
-                output_mode='json'
-            )['body'].read()
+        )['body'].read()
+        tasks = json.loads(taskresp)['entry']
 
-            raise Exception(
-                'Error rotating HEC token {} on Splunk host '
-                '{}. Input was not found! Input list: {}'.format(
-                    inputname,
-                    self._service.host,
-                    inputlist
+        if tasks[0]['name'] == 'lastDeploy':
+            return tasks[0]['content']['taskId']
+        else:
+            raise Exception('Could not fetch last task Id!')
```

Comment:
Do we know why? Is it in the response?

Reply:
This should not happen, and would be a strange scenario where the call was successful but the response did not match the documented format. At any rate, fair point, updated to be slightly more helpful in troubleshooting.